python/paddle/tensor/math.py

```diff
@@ -125,64 +127,76 @@
+# @templatedoc()
+def power(x, exponent, name=None):
     """
-	:alias_main: paddle.pow
-	:alias: paddle.pow,paddle.tensor.pow,paddle.tensor.math.pow
-
-    This is Pow Activation Operator.
-
-    :math:`out = input^{exponent}`
-
-    Args:
-        input(Variable): A ``Tensor`` or ``LoDTensor`` . The data type is ``float32`` or ``float64``.
-        exponent(float32|Variable): A scalar with type ``float32`` or a ``Tensor`` with shape [1] and type ``float32``.
-        name(str, optional): The default value is None. Normally there is no need for user to set this property.
-            For more information, please refer to :ref:`api_guide_Name` .
-
+	:alias_main: paddle.power
+	:alias: paddle.power,paddle.tensor.power,paddle.tensor.math.power
+    :math: ``out = x^{exponent}``    
+    x(Tensor): A ``Tensor`` with type ``float32`` or ``float64``.
+    exponent(float32|Tensor): A scalar with type ``float32`` or a ``Tensor`` with shape [1] and type ``float32``.
+    name(str, optional): The default value is None. Normally there is no need for user to set this property. 
+    For more information, please refer to :ref:`api_guide_Name` .
     Returns:
-        Variable: A ``Tensor`` or ``LoDTensor``. The data type is same as ``input``.
-
+        Tensor: A ``Tensor`` with type same as ``x``.
     Examples:
-
         .. code-block:: python
-
             import paddle
-            import paddle.fluid as fluid
-
-            x = fluid.data(name="x", shape=[32,32], dtype="float32")
-
-            # example 1: argument exponent is float
-            y_1 = paddle.pow(x, 2.0)
-            # y_1 is x^{2.0}
-
-            # example 2: argument exponent is Variable
-            exponent_tensor = fluid.layers.fill_constant([1], "float32", 3.0)
-            y_2 = paddle.pow(x, exponent_tensor)
-            # y_2 is x^{3.0}
+            import numpy as numpy
+            paddle.enable_imperative()
+            # example 1: exponent is a float
+            x_data = np.array([1, 2, 3])
+            exponent = 2
+            x = paddle.imperative.to_variable(x_data)
+            res = paddle.power(x, exponent)
+            print(res.numpy()) # [1 4 9]
+            
+            # example 2: exponent is a Variable
+            exponent = paddle.fill_constant(shape=[1], value=2, dtype='float32')
+            res = paddle.power(x, exponent)
+            print(res.numpy()) # [1 4 9]
     """
+    # in dynamic graph mode
     if in_dygraph_mode():
-        return core.ops.pow(input, "exponent", exponent)
-
-    helper = LayerHelper('pow', **locals())
-    inputs = {'X': input}
-    attrs = {}
-    if isinstance(exponent, Variable):
-        exponent.stop_gradient = True
-        inputs['FactorTensor'] = exponent
+        # exponent is scalar
+        if isinstance(exponent, (int, float)):
+            return core.ops.pow(x, 'factor', exponent)
+        # exponent is Tensor
+        else:
+            # if check_dtype(exponent.dtype, 'exponent', ['int32', 'int64']):
```

Comment:
1. You can add if check like your code below: x.dtype != exponent.dtype
2. You can remove useless code in the future
3. Usually when you don't need to cast (for example, both x and exponent are float32, you should return float32) here your code will always return float64.

I heard paddle API 2.0 standard will be as close to numpy as you can. I don't think here you are returning same types as numpy, below is what I tried on python numpy:

![image](https://user-images.githubusercontent.com/7913861/90463867-97d8a100-e13e-11ea-969b-6af04781ed95.png)